spinnaker_graph_front_end/examples/hello_world_untimed/src/hello_world.c

```diff
@@ -0,0 +1,179 @@
+    bool success = recording_initialize(&recording_region, &recording_flags);
+    return success;
+}
+
+static void resume_callback(void);
+
+static void run(uint unused0, uint unused1) {
+    use(unused0);
+    use(unused1);
+
+    // Increment the step as it will start at "-1"
+    step++;
+
+    log_info("Running from step %u until %u or %u", step, n_steps, c_data->max_index);
+    while ((run_forever || step < n_steps) && step < c_data->max_index) {
+        if (recording_flags) {
+            record_data();
+            recording_do_timestep_update(step);
+        }
+        step++;
+    }
+
+    log_info("Steps run");
+    simulation_handle_pause_resume(resume_callback);
+    if (recording_flags) {
+        recording_finalise();
+    }
+    log_info("Making ready...");
+    simulation_ready_to_read();
+    log_info("Done");
+}
+
+// ! \brief Called when simulation is about to restart
+static void resume_callback(void) {
+    recording_reset();
+    log_info("Scheduling callback from resume");
+    spin1_schedule_callback(run, 0, 0, 1);
+}
+
+
+static void exit_callback() {
+    if (recording_flags) {
+        recording_finalise();
+    }
+}
+
+static void start_callback() {
+    log_info("Scheduling callback from start");
+    spin1_schedule_callback(run, 0, 0, 1);
+}
+
+static bool initialize() {
+    // Get the address this core's DTCM data starts at from SRAM
+    data_specification_metadata_t *data = data_specification_get_data_address();
+
+    // Read the header
+    if (!data_specification_read_header(data)) {
+        log_error("failed to read the data spec header");
+        return false;
+    }
+
+    // set up the simulation interface
+    uint32_t unused_timer_period;
+    if (!simulation_initialise(
+            data_specification_get_region(SYSTEM_REGION, data),
+            APPLICATION_NAME_HASH, &unused_timer_period, &n_steps,
+            &run_forever, &step, SDP, DMA)) {
+        return false;
+    }
+
+    char_data *sdram_data = data_specification_get_region(PARAMS_REGION, data);
+    uint32_t size = sizeof(char_data) + (sdram_data->n_chars * sizeof(uint8_t));
+    c_data = spin1_malloc(size);
+    if (c_data == NULL) {
+        log_error("Failed to allocate local data");
+        return false;
+    }
+    spin1_memcpy(c_data, sdram_data, size);
+    return true;
+}
+
+/****f*
+ *
+ * SUMMARY
+ *  This function is called at application start-up.
+ *  It is used to register event callbacks and begin the simulation.
+ *
+ * SYNOPSIS
+ *  int c_main()
+ *
+ * SOURCE
+ */
+void c_main(void) {
+    // initialise the model
+    if (!initialize()) {
+        rt_error(RTE_SWERR);
+    }
+
+    // initialise the recording section
+    // set up recording data structures
+    if (!initialise_recording()) {
+         rt_error(RTE_SWERR);
+    }
+
+    simulation_set_exit_function(exit_callback);
+    simulation_set_start_function(start_callback);
+    simulation_set_uses_timer(0);
```

Comment:
this def shows how its really a bool. heck. why not support uses and does_not_use timer?

Reply:
Updated